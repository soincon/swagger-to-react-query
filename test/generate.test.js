const generator = require('../src/generator');

test('generate', async () => {
    const config = {
        name: 'My API',
        input: {
            json: require('./apiSchema.json'),
        },
        output: {
            dir: './generated/snapshotTest',
        },
        apiUrl: {
            value: "'http://my-api.com'",
        },
    };

    const specs = require('./apiSchema.json');
    const { controllers, globalTypes } = await generator({ specs, config });
    expect(globalTypes).toMatchSnapshot();
    Object.entries(controllers).forEach(([, { types }]) => {
        expect(types).toMatchSnapshot();
    });
});
