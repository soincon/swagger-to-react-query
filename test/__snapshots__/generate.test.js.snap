// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate 1`] = `
"/* eslint-disable */
/* tslint:disable */
export type CountryDto = { versionLock?: number; active?: boolean; createdAt?: string; modifiedAt?: string; modifiedBy?: number; id?: number; name: string; isoCode: string; description?: string; links?: Link[] }

export type CountryFilter = { active?: boolean; id?: number; ids?: number[]; name?: string; isoCode?: string; description?: string }

export type PageDto = { content?: {}[]; page?: number; size?: number; totalElements?: number; empty?: boolean; first?: boolean; last?: boolean; numberOfElements?: number; totalPages?: number }


"
`;

exports[`generate 2`] = `
"/* eslint-disable */
/* tslint:disable */
import { UseQueryOptions, UseQueryResult, UseMutationOptions, MutateFunction, UseMutationResult } from 'react-query'
import { RequestInit } from 'node-fetch'

export type CountryDto = { versionLock?: number; active?: boolean; createdAt?: string; modifiedAt?: string; modifiedBy?: number; id?: number; name: string; isoCode: string; description?: string; links?: Link[] }

export type CountryFilter = { active?: boolean; id?: number; ids?: number[]; name?: string; isoCode?: string; description?: string }

export type PageDto = { content?: {}[]; page?: number; size?: number; totalElements?: number; empty?: boolean; first?: boolean; last?: boolean; numberOfElements?: number; totalPages?: number }

export type GetCountryPathParams = { clientId?: number; id: number }
/**
 * Returns the specified country

 * Retrieves the data that match with the specified identifier

 * 
 */
export function getCountry(pathParams?: GetCountryPathParams, options?: RequestInit): Promise<CountryDto>;
/**
 * Returns the specified country

 * Retrieves the data that match with the specified identifier

 * 
 */
export function useGetCountry(pathParams?: GetCountryPathParams, config?: UseQueryOptions<CountryDto>, options?: RequestInit): UseQueryResult<CountryDto>;
declare namespace useGetCountry {
    /** Entire key: \`['{apiUrl}/v1/{clientId}/countries/{id}', pathParams]\` */
    export const queryKey: string;
}

export type PutCountryPathParams = { clientId?: number; id: number }
/**
 * Updates the specified country

 * Updates the existing data with the specified one in the request body

 * 
 */
export function putCountry(pathParams?: PutCountryPathParams, body?: CountryDto | [PutCountryPathParams?, void?, CountryDto?, RequestInit?], options?: RequestInit): Promise<CountryDto>;
/**
 * Updates the specified country

 * Updates the existing data with the specified one in the request body

 * 
 */
export function usePutCountry(pathParams?: PutCountryPathParams, config?: UseMutationOptions<CountryDto, unknown, CountryDto>, options?: RequestInit): UseMutationResult<CountryDto, unknown, CountryDto>;
export type DeleteCountryPathParams = { id: number }
/**
 * Deletes permanently the specified country

 * Removes the existing data permanently

 * 
 */
export function deleteCountry(pathParams?: DeleteCountryPathParams, options?: RequestInit): Promise<void>;
/**
 * Deletes permanently the specified country

 * Removes the existing data permanently

 * 
 */
export function useDeleteCountry(pathParams?: DeleteCountryPathParams, config?: UseMutationOptions<void, unknown, void>, options?: RequestInit): UseMutationResult<void, unknown, void>;
export type PatchCountryPathParams = { id: number }
/**
 * Updates 'partially' the specified country

 * Updates the existing data with a structure which has the changes to apply (see rfc6902)

 * 
 */
export function patchCountry(pathParams?: PatchCountryPathParams, body?: JsonPatch | [PatchCountryPathParams?, void?, JsonPatch?, RequestInit?], options?: RequestInit): Promise<CountryDto>;
/**
 * Updates 'partially' the specified country

 * Updates the existing data with a structure which has the changes to apply (see rfc6902)

 * 
 */
export function usePatchCountry(pathParams?: PatchCountryPathParams, config?: UseMutationOptions<CountryDto, unknown, JsonPatch>, options?: RequestInit): UseMutationResult<CountryDto, unknown, JsonPatch>;
/**
 * Creates a new country

 * Creates a new country using the specified data

 * 
 */
export function postCountry(body?: CountryDto | [void?, void?, CountryDto?, RequestInit?], options?: RequestInit): Promise<CountryDto>;
/**
 * Creates a new country

 * Creates a new country using the specified data

 * 
 */
export function usePostCountry(config?: UseMutationOptions<CountryDto, unknown, CountryDto>, options?: RequestInit): UseMutationResult<CountryDto, unknown, CountryDto>;
export type PostCountryWithPathParamsPathParams = { clientId?: number; someParam: number }
/**
 * Creates a new country

 * Creates a new country using the specified data

 * 
 */
export function postCountryWithPathParams(pathParams?: PostCountryWithPathParamsPathParams, body?: CountryDto | [PostCountryWithPathParamsPathParams?, void?, CountryDto?, RequestInit?], options?: RequestInit): Promise<CountryDto>;
/**
 * Creates a new country

 * Creates a new country using the specified data

 * 
 */
export function usePostCountryWithPathParams(pathParams?: PostCountryWithPathParamsPathParams, config?: UseMutationOptions<CountryDto, unknown, CountryDto>, options?: RequestInit): UseMutationResult<CountryDto, unknown, CountryDto>;
export type GetCountryWithQueryPathParams = { clientId?: number; id: number }
export type GetCountryWithQueryQueryParams = { param?: number }
/**
 * 
 * 
 * 
 */
export function getCountryWithQuery(pathParams?: GetCountryWithQueryPathParams, queryParams?: GetCountryWithQueryQueryParams, options?: RequestInit): Promise<CountryDto>;
/**
 * 
 * 
 * 
 */
export function useGetCountryWithQuery(pathParams?: GetCountryWithQueryPathParams, queryParams?: GetCountryWithQueryQueryParams, config?: UseQueryOptions<CountryDto>, options?: RequestInit): UseQueryResult<CountryDto>;
declare namespace useGetCountryWithQuery {
    /** Entire key: \`['{apiUrl}/v1/countries/search/{id}', pathParams, queryParams]\` */
    export const queryKey: string;
}

export type SearchCountriesPathParams = { id?: number }
export type SearchCountriesQueryParams = { page?: number; size?: number; sort?: string[] }
/**
 * Returns all countries which match with the specified filter

 * Retrieves every country for the specified filter

 * 
 */
export function searchCountries(pathParams?: SearchCountriesPathParams, body?: CountryFilter | [SearchCountriesPathParams?, SearchCountriesQueryParams?, CountryFilter?, RequestInit?], queryParams?: SearchCountriesQueryParams, options?: RequestInit): Promise<PageDto>;
/**
 * Returns all countries which match with the specified filter

 * Retrieves every country for the specified filter

 * 
 */
export function useSearchCountries(pathParams?: SearchCountriesPathParams, body?: CountryFilter, queryParams?: SearchCountriesQueryParams, config?: UseQueryOptions<PageDto>, options?: RequestInit): UseQueryResult<PageDto>;
declare namespace useSearchCountries {
    /** Entire key: \`['{apiUrl}/v1/countries/search/{id}', pathParams, body, queryParams]\` */
    export const queryKey: string;
}

"
`;

exports[`generate 3`] = `
"/* eslint-disable */
/* tslint:disable */
import { UseQueryOptions, UseQueryResult, UseMutationOptions, MutateFunction, UseMutationResult } from 'react-query'
import { RequestInit } from 'node-fetch'

export type CountryDto = { versionLock?: number; active?: boolean; createdAt?: string; modifiedAt?: string; modifiedBy?: number; id?: number; name: string; isoCode: string; description?: string; links?: Link[] }

export type CountryFilter = { active?: boolean; id?: number; ids?: number[]; name?: string; isoCode?: string; description?: string }

export type PageDto = { content?: {}[]; page?: number; size?: number; totalElements?: number; empty?: boolean; first?: boolean; last?: boolean; numberOfElements?: number; totalPages?: number }

/**
 * Creates a new foo

 * Creates a new country using the specified data

 * 
 */
export function postFooBar(options?: RequestInit): Promise<CountryDto>;
/**
 * Creates a new foo

 * Creates a new country using the specified data

 * 
 */
export function usePostFooBar(config?: UseQueryOptions<CountryDto>, options?: RequestInit): UseQueryResult<CountryDto>;
declare namespace usePostFooBar {
    /** Entire key: \`['{apiUrl}/v1/foo', ]\` */
    export const queryKey: string;
}

"
`;
